applications/serial_lte_modem/doc/slm_description.rst

```diff
@@ -0,0 +1,818 @@
+         #XSEND: 15
+         OK
+
+         **AT#XRECV**
+         PONG: b'Test TLS client'
+         #XRECV: 1, 24
+         OK
+
+   #. Close the socket.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=0**
+         #XSOCKET: 0, closed
+         OK
+
+#. Test a TLS client with TCP proxy service (note that these commands are available only if :option:`CONFIG_SLM_TCP_PROXY` is defined):
+
+   a. Create a TCP/TLS client and connect to a server.
+      Replace *example.com* with the host name or IPv4 address of a TLS server and *1234* with the corresponding port.
+      Then read information about the connection.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XTCPCLI=1,"**\ *example.com*\ **",**\ *1234*
+         #XTCPCLI: 2 connected
+         OK
+
+         **AT#XTCPCLI?**
+         #XTCPCLI: 1
+         OK
+
+   #. Send plain text data to the TLS server and retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XTCPSEND=1,"Test TLS client"**
+         #XTCPSEND: 15
+         OK
+         #XTCPDATA: 1, 24
+
+         **AT#XTCPRECV**
+         PONG: b'Test TLS client'
+         #XTCPRECV: 24
+         OK
+
+   #. Disconnect from the server.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XTCPCLI=0**
+         #XTCPCLI: disconnected
+         OK
+
+DTLS client
+~~~~~~~~~~~
+
+The DTLS client requires connection-based UDP to trigger the DTLS establishment.
+
+Before completing this test, you must update the Pre-shared Key (PSK) and the PSK identity to be used for the TLS connection in the modem.
+The credentials must use the security tag 16842756.
+
+To store the credentials in the modem, enter the following AT commands:
+
+.. parsed-literal::
+   :class: highlight
+
+   **AT%CMNG=0,16842756,3,"6e7266393174657374"**
+   **AT%CMNG=0,16842756,4,"nrf91test"**
+
+You must register the same PSK and PSK identity on the server side.
+
+1. Establish and test a DTLS connection:
+
+   a. List the credentials that are stored in the modem with security tag 16842755:
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT%CMNG=1,16842756**
+         %CMNG: 16842756,3,"0303030303030303030303030303030303030303030303030303030303030303"
+         %CMNG: 16842756,4,"0404040404040404040404040404040404040404040404040404040404040404"
+         OK
+
+   #. Open a TCP/DTLS socket that uses the security tag 16842756 and connect to a DTLS server on a specified port.
+      Replace *example.com* with the host name or IPv4 address of a DTLS server and *1234* with the corresponding port.
+
+     .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=1,2,0,16842756**
+         #XSOCKET: 2, 2, 0, 273
+         OK
+
+         **AT#XCONNECT="**\ *example.com*\ **",**\ *1234*
+         #XCONNECT: 1
+         OK
+
+   #. Send plain text data to the DTLS server and retrieve the returned data.
```

Comment:
This step was missing in the Word document, but we should probably send data to test?
If so, is it correct?

Reply:
I've never been able to test against a DTLS server, only DTLS handshake (i.e. connect) so far.
@lats1980 maybe we could try openssh on Contabo later, run as DTLS server there

Reply:
Should I remove the step for now?

Reply:
Yes, please remove DTLS client description for now. Another known issue is the DTLS client has problem with certificate authentication on the modem side. I only tested PSK-based authentication

Reply:
Removed.